ecommerce/extensions/payment/processors/paystack.py

```diff
@@ -0,0 +1,199 @@
+                'value': product.course_id
+            },
+            {
+                'display_name': 'Course Name',
+                'variable_name': 'course_title',
+                'value': product.course.name
+            }
+        ]
+
+    def get_transaction_parameters(self, basket, request=None, use_client_side_checkout=True, **kwargs):
+        """
+        Get Payment form URL from paystack.
+
+        Arguments:
+            basket (Basket): The basket of products being purchased.
+            request (Request, optional): A Request object which is used to construct Paystack's `return_url`.
+            use_client_side_checkout (bool, optional): This value is not used.
+            **kwargs: Additional parameters; not used by this method.
+
+        Returns:
+            dict: paystack-specific parameters required to complete a transaction. Must contain a URL
+                of payment form to which users can be directed in order to approve a newly created payment.
+
+        Raises:
+            GatewayError: Indicates a general error or unexpected behavior on the part of PayPal which prevented
+            a payment from being created.
+        """
+        data = {
+            'amount': self.get_basket_amount(basket),
+            'email': basket.owner.email,
+            'callback_url': self.return_url,
+            'metadata': {
+                'cancel_action:': self.cancel_url,
+                'order_number': basket.order_number,
+                'basket_id': basket.id,
+                'custom_fields': self.get_paystack_custom_fields_data(basket)
+            },
+        }
+
+        success, response = self.paystack_client.handler(INITIALIZE_TRANSACTION_CODE, data)
+
+        if success:
+            logger.info("Successfully got hosted Paystack payment page for basket [%d].", basket.id)
+            data = response.get('data')
+            if data:
+                return {'payment_page_url': data.get('authorization_url')}
+
+        logger.error("Failed to get Paystack payment form for basket [%d].", basket.id)
+        raise GatewayError("Paystack payment creation failure: unable to get Paystack form token")
+
+    def handle_processor_response(self, response, basket=None):
+        """
+        Execute an approved Paystack transaction. This method will record payment processor
+        response for future usage.
+
+        Arguments:
+            response: Transaction response received from Paystack after successfull payment
+            basket (Basket): Basket being purchased via the payment processor.
+
+        Returns:
+            HandledProcessorResponse
+        """
+        authorization = response.get('authorization')
+        transaction_id = response.get('id')
+
+        self.record_processor_response(authorization, transaction_id=transaction_id, basket=basket)
+        logger.info("Successfully executed Paystack payment [%s] for basket [%d].", transaction_id, basket.id)
```

Comment:
can this method can be called via `basket=None`

Reply:
answered above